pkg/database/postgresql.go

```diff
@@ -27,11 +27,31 @@ func (info PostgreSQLDatabaseInfo) GetConnectionString() string {
 }
 
 func (info PostgreSQLDatabaseInfo) GetResourcesSQL() string {
-	return "SELECT data FROM resource WHERE kind=$1 AND api_version=$2"
+	return "SELECT data->'metadata'->'creationTimestamp', uuid, data FROM resource WHERE kind=$1 AND api_version=$2 ORDER BY ((data->'metadata'->'creationTimestamp')::text::timestamptz, uuid)"
+}
+
+func (info PostgreSQLDatabaseInfo) GetResourcesLimitedSQL() string {
```

Comment:
The number of routines here is exploding and I fear will be hard to maintain. There's also the duplication of lots of SQL in each of these queries.

It might make sense to go back to a single routine and have all the SQL generated behind the scenes.  Instead, the user would have to pass a map or struct containing parameter values, which they can change or leave empty to be defaulted.  The routine can then generate the SQL to be used.

I think this would 1) cut down the method bloat we see here, 2) reduce duplication of SQL,  and 3) simplify things at the call site. Generating the SQL itself might be a little tricky, but it should be able to be done as the SQL statement itself has to be well-structured.

This might be even easier if go allowed for default values. But it doesn't, so something else needs to be done here. Here's [one blog](https://joneisen.me/programming/2013/06/23/golang-and-default-values.html) with some suggestions.

Reply:
This was already discussed in https://github.com/kubearchive/kubearchive/pull/484#discussion_r1795734437 and I opened an issue to start addressing refactorings here: https://github.com/kubearchive/kubearchive/issues/537

Reply:
Why are we planning on refactoring code that hasn't even been committed yet? That just seems wrong.

I don't think this feature is a rush, so I'd rather get it right the first time rather than commit it knowing we are already going to change it.

Reply:
ok, now that I am rereading your first comment in this thread, I see that I wasn't talking about the same.

I complained about duplicated code in the `database.go` file, but you are talking explicitly about the SQL sentences (my bad).

That means that my previous comment doesn't apply, and BTW, we created a separate issue because what is proposed is to change code that was already there, not new code.

So, going back to your first suggestion and now that I am focused on the problem that you mentioned: having too many routines with different SQL sentences

> It might make sense to go back to a single routine and have all the SQL generated behind the scenes. Instead, the user would have to pass a map or struct containing parameter values, which they can change or leave empty to be defaulted. The routine can then generate the SQL to be used.

I am not sure of how this will look like.
Now the approach is more explicit, and I usually like that. I also like how with different database integrations just need to worry about implementing these routines.

I can't say anything "bad" about your proposal because I don't envision it right now, and that makes me wonder if it should be in this PR or not.

This PR just keeps going with the last design that we did, refactoring that wouldn't be part of the scope of the pagination issue and could get this PR harder to review.

> There's also the duplication of lots of SQL in each of these queries.

How is this? every SQL statement in this file is different from each other, of course, there are duplication at some parts of the query (mostly, this part: `SELECT data->'metadata'->'creationTimestamp', uuid FROM resource WHERE kind=$1 AND api_version=$2`) but getting this part separated from the rest of the query I think that just get things less readable and maintainable

Reply:
> This PR just keeps going with the last design that we did,

Where is this design?

Reply:
Implementing the pagination I'm just following the same patterns we had, so in a way the code will need to be refactored after merging the pagination feature. I can do the refactor here, but the PR will just get bigger and it will more difficult to review.

Reply:
> Where is this design?

I was referring to using the same patterns that we had, the ones that were defined in https://github.com/kubearchive/kubearchive/pull/469